co-log/src/Colog/Actions.hs

```diff
@@ -92,9 +102,37 @@ withLogRotation
     -> (LogAction m msg -> IO r)
     -- ^ Continuation action
     -> IO r
-withLogRotation sizeLimit filesLimit path cleanup mkAction cont = cont rotationAction
+withLogRotation sizeLimit filesLimit path cleanup mkAction cont = do
+    folderStatus <- getLogFolderStatus path
+    let (toRename, toCleanup) = getLogActionFiles sizeLimit filesLimit path folderStatus
+
+    mapM_ cleanup toCleanup
+    mapM_ (D.renameFile path) toRename
+
+    cont rotationAction
   where
     rotationAction :: LogAction m msg
-    rotationAction = LogAction $ \msg -> do
-        withFile path AppendMode writeFileLoop
--}
+    rotationAction = liftLogIO $ LogAction $ (\msg ->
+        withFile path AppendMode (flip (unLogAction . mkAction) msg))
```

Comment:
You got the idea right. But returns `rotationAction` is not valid, unfortunately. You return this action and it's supposed to do all the dirty work. So on every call to `LogAction` this function should output line to the file, check resulted length of the file and rename files if needed. Currently you're performing renaming only once, at the beginning of `withLogRotation` while in reality this should be done on every call to `rotationAction`.

The difficulty in implementing this function is checking the size of the file each time, keep outputting to `Handle` instead of reopening file each time and use new `Handle` each time the size of the file is bigger than the given `Limit`

Reply:
> Would we expect the user to have a partially applied withLogRotation and pass cont each time they want to log?

The use-case is that you call `withLogRotation` only once in your program to create properly configured `LogAction` and then you use only that action. Compare with more simple example: `withFile`. It's more efficient to open file only once and use `Handle` to write output instead of calling `appendFile` on every message. So you open file only once in your program, and then you use given action to output the result. Same with `withLogRotation`. You specify configuration parameters and then you can use single action to output messages, and this action handles file renaming and size check for you.